File: frontend/src/objects/submission/request/SubmissionUserQuery.ts
/** 提交列表用户搜索词品牌类型；表示已确认非空的用户过滤输入。 */
export type SubmissionUserQuery = string & { readonly __brand: 'SubmissionUserQuery' }

/** 解析 helper 的返回结果；成功携带品牌值，失败携带可展示错误，不抛异常。 */
type ParseResult<T> = { ok: true; value: T } | { ok: false; error: string }

/** 创建提交用户搜索词品牌值；调用前必须确认非空。 */
function createSubmissionUserQuery(value: string): SubmissionUserQuery {
  /** 注意：这里的 as 只在 parseSubmissionUserQuery 校验通过后施加品牌类型。 */
  return value as SubmissionUserQuery
}

/** 将提交用户搜索词还原为查询字符串；无副作用。 */
export function submissionUserQueryValue(query: SubmissionUserQuery): string {
  return query
}

/** 解析提交用户搜索输入；去除首尾空白并拒绝空查询。 */
export function parseSubmissionUserQuery(rawQuery: string): ParseResult<SubmissionUserQuery> {
  const normalized = rawQuery.trim()
  if (!normalized) {
    return { ok: false, error: 'Submission username query is required.' }
  }
  return { ok: true, value: createSubmissionUserQuery(normalized) }
}

File: backend/src/main/scala/domains/submission/objects/request/SubmissionUserQuery.scala
package domains.submission.objects.request

import io.circe.{Decoder, Encoder}


/** 提交列表中的用户搜索词，可匹配用户名或展示名。 */
final case class SubmissionUserQuery(value: String)

/** 用户搜索词 JSON 编解码与非空校验入口。 */
object SubmissionUserQuery:
  given Encoder[SubmissionUserQuery] = Encoder.encodeString.contramap(_.value)
  given Decoder[SubmissionUserQuery] = Decoder.decodeString.emap(parse)

  /** 规范化用户搜索词；空白输入视为错误。 */
  def parse(raw: String): Either[String, SubmissionUserQuery] =
    val normalized = raw.trim
    if normalized.isEmpty then Left("Submission username query is required.")
    else Right(SubmissionUserQuery(normalized))
